flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/OneInputStreamTaskTest.java

```diff
@@ -1011,5 +1065,70 @@ protected void handleWatermark(Watermark mark) {
 			output.emitWatermark(mark);
 		}
 	}
+
+	/**
+	 * Uses to test handling the EndOfInput notification.
+	 */
+	static class TestBoundedOneInputOperator extends AbstractStreamOperator<String>
+		implements OneInputStreamOperator<String, String>, BoundedOneInput {
+
+		private static final long serialVersionUID = 1L;
+
+		private final String name;
+
+		private volatile int outputRecords;
+
+		private static final CompletableFuture<Void> success = CompletableFuture.completedFuture(null);
+		private static final CompletableFuture<Void> failure = FutureUtils.completedExceptionally(
+			new Exception("Not in line with expectation."));
+
+		public TestBoundedOneInputOperator(String name) {
+			this.name = name;
+		}
+
+		@Override
+		public void processElement(StreamRecord<String> element) {
+			output.collect(element);
+			outputRecords++;
+
+			synchronized (this) {
+				this.notifyAll();
+			}
+		}
+
+		@Override
+		public void endInput() {
+			output.collect(new StreamRecord<>("[" + name + "]: Bye"));
+			outputRecords++;
+
+			synchronized (this) {
+				this.notifyAll();
+			}
+		}
+
+		public void waitOutputRecords(int expectedRecords) throws Exception {
```

Comment:
Why do you need this method?

Reply:
This test was written a long time ago, when it was needed. Now not needed, and I will remove it.